### PR TITLE
Rework flaky tests in user search spec

### PR DESCRIPTION
**Pull request by felixauringer**
This uses predefined users (via a FactoryBot trait) in the user search spec (`spec/features/user_search_spec.rb`) so that these specs do not fail some of the time due to bad luck with the random names of the default user factory.  
The problem was mentioned [here](https://github.com/hpi-swt2/connections-portal/pull/100#discussion_r559185070) and [here](https://github.com/hpi-swt2/connections-portal/pull/150#issuecomment-765705962).  

I now made sure that all strings that are searched for occur exactly in one user in the tested attribute.

### CODE REVIEW

**Comment by frcroth** [comment]
@felixauringer Is this PR still valid? If yes, can it be merged?

**Comment by felixauringer** [comment]
> @felixauringer Is this PR still valid? If yes, can it be merged?

In my opinion it is still relevant although the tests fail only rarely. It can be merged after it is reviewed, I added some reviewers now :sweat_smile:

File: spec/features/user_search_spec.rb [hunk 1]
```diff
@@ -2,50 +2,75 @@
...
-    visit search_users_path(search: second_user.lastname[0..2])
-    expect(page).to have_text(second_user.lastname)
-  end
+    before { visit search_users_path }
 
-  it 'is possible to filter by username' do
-    visit search_users_path(search: second_user.username[0..2])
-    expect(page).to have_text(second_user.username)
-  end
+    it 'does show all other users when no filter is given' do
```
Comment (frcroth) [inline spec/features/user_search_spec.rb]:
```suggestion
    it 'shows all other users when no filter is given' do
```

**Review by frcroth — APPROVED** [review]

File: spec/factories/users.rb [hunk 2]
```diff
@@ -16,6 +16,20 @@
...
+      lastname { 'Mustermann' }
+      username { 'standard_user' }
+      email { 'test@example.com' }
+    end
+
+    trait :erika do
+      firstname { 'Erika' }
+      lastname { 'Musterfrau' }
+      username { 'erika.musterfrau' }
+      email { 'erika.musterfrau@example.com' }
```
Comment (Paula-Kli) [inline spec/factories/users.rb]:
```suggestion
      email { 'test2@example.com' }
```

**Review by Paula-Kli — APPROVED** [review]

**Pull request merged**